content/building-blocks/automate-and-execute-your-work/reproducible-work/config-VM-GCP.md

```diff
@@ -0,0 +1,326 @@
+authorlink: "https://www.linkedin.com/in/fernando-iscar/"
+draft: false
+date: 2023-06-05
+aliases: 
+  - /run/vm-on-google-cloud
+  - /handle/memory-issues
+  - /import/heavy-files
+---
+
+# Gear up your code with Google Cloud VMs
+
+Tired of encountering memory or power constraints while running your code or waiting long due to your machine's limited capacity? In this building block, you'll discover how to create and configure a simple and robust VM instance in **Google Cloud**, designed to overcome these limitations. Say goodbye to obstacles and embrace seamless computing with this comprehensive guide.
+
+## Initialize a new instance
+
+### Create a Google Cloud account
+
+First of all, we will need to have a [Google Cloud](https://cloud.google.com/?hl=en) account and a project created in order to create an instance. After this is done, we can go to the welcome page and click on **Create a VM**.
+
+<p align = "center">
+<img src = "../img/welcome.png" width="700" style="border:1px solid black;">
+<figcaption> Click the button to create a new instance</figcaption>
+</p>
+
+### Set up your Virtual Machine
+
+You will be directed to the instance configuration page. Here, you will need to enter details for your instance. This includes the **instance name**, **labels**, **region**, **zone**, and **machine configuration**.
+
+You'll encounter four primary machine categories to select from:
+
+- **General-purpose:** Offers the best balance between price and performance, making it suitable for a wide range of workloads. It provides a cost-effective solution without compromising efficiency.
+
+- **Compute-optimized:** Delivers the highest performance per core on Compute Engine, specifically designed for compute-intensive tasks. It excels in scenarios that require substantial computational power, ensuring faster processing and reduced execution times.
+
+- **Memory-optimized:** Specifically designed for memory-intensive workloads, this category offers a higher memory capacity per core compared to other machine families. With the ability to support up to 12 TB of memory, it empowers applications that heavily rely on data storage and manipulation, enabling efficient data processing and analysis.
+
+- **Accelerator-optimized:** The perfect choice for massively parallelized Compute Unified Device Architecture ([CUDA](https://developer.nvidia.com/cuda-zone#:~:text=CUDA%C2%AE%20is%20a%20parallel,harnessing%20the%20power%20of%20GPUs.)) compute workloads, such as machine learning and high-performance computing. This category is specifically tailored for workloads that demand GPU acceleration, ensuring exceptional performance for tasks that require substantial graphical processing power and parallel computing capabilities.
+
+A good choice to beat memory hitches and supercharge processing could be going with an **NVIDIA L4 g2-standard-32** machine. It's packed with 128GB of RAM and a GPU, perfect for heavyweight computing, but still cost-friendly.
```

Comment:
you can add a note here that a GPU enabled VM is needed mainly if someone plans on doing deep learning tasks (e.g language models) - for other use cases a GPU is not needed and may add to VM costs. this is also a good resource to link, which shows that using GPUs suboptimally can even add to compute time: https://rstudio-pubs-static.s3.amazonaws.com/15192_5965f6c170994ebb972deaf18f1ddf34.html